swarm/api/api.go

```diff
@@ -256,33 +252,37 @@ func (self *Api) Resolve(uri *URI) (storage.Key, error) {
 	apiResolveCount.Inc(1)
 	log.Trace("resolving", "uri", uri.Addr)
 
-	// if the URI is immutable, check if the address is a hash
-	isHash := hashMatcher.MatchString(uri.Addr)
+	// if the URI is immutable, check if the address looks like a hash
 	if uri.Immutable() || uri.DeprecatedImmutable() {
-		if !isHash {
+		key := uri.Key()
```

Comment:
We can take `key := uri.Key()ʼ out of the ifs to avoid repeating it, but there is a case where you'd be parsing the key and not using it. In any case, this code does the same thing as the previous code, where parsing the key was part of the return statement.